kafka-agent/src/main/java/io/strimzi/kafka/agent/KafkaAgent.java

```diff
@@ -65,28 +136,26 @@ public void onMetricRemoved(MetricName metricName) {
             @Override
             public synchronized void onMetricAdded(MetricName metricName, Metric metric) {
                 LOGGER.trace("Metric added {}", metricName);
-                if ("kafka.server".equals(metricName.getGroup())) {
-                    if ("KafkaServer".equals(metricName.getType())
-                            && "BrokerState".equals(metricName.getName())
-                            && metric instanceof Gauge) {
-                        LOGGER.debug("Metric {} added ", metricName);
-                        brokerStateName = metricName;
-                        brokerState = (Gauge) metric;
-                    } else if ("SessionExpireListener".equals(metricName.getType())
-                            && "SessionState".equals(metricName.getName())
-                            && metric instanceof Gauge) {
-                        sessionStateName = metricName;
-                        sessionState = (Gauge) metric;
-                    }
+                if (isBrokerState(metricName) && metric instanceof Gauge) {
+                    LOGGER.debug("Metric {} added ", metricName);
+                    brokerStateName = metricName;
+                    brokerState = (Gauge) metric;
+                } else if (isRemainingLogsToRecover(metricName) && metric instanceof Gauge) {
+                    LOGGER.debug("Metric {} added ", metricName);
+                    remainingLogsToRecover = (Gauge) metric;
+                } else if (isRemainingSegmentsToRecover(metricName) && metric instanceof Gauge) {
+                    LOGGER.debug("Metric {} added ", metricName);
+                    remainingSegmentsToRecover = (Gauge) metric;
+                } else if (isSessionState(metricName)
```

Comment:
do we need all the above `LOGGER.debug` to log a metric was added? I see a `LOGGER.trace("Metric added {}", metricName);` at the top which is logging for all of them.

Reply:
I guess it was for getting different level of loggings, trace being more verbose than debug which is just logging broker metrics. I'm happy to remove it anyway, as metrics get logged either way.

Reply:
We could change the common one to be `debug` then. Not sure why it was trace.